core/components/organisms/verticalMenu/VerticalMenu.tsx

```diff
@@ -0,0 +1,205 @@
+import { BaseProps, extractBaseProps } from '@/utils/types';
+import {
+  getMenu,
+  isMenuActive,
+  ActiveMenu,
+  Menu
+} from '../navigation/utils';
+
+export interface VerticalMenuProps extends BaseProps {
+  /**
+   * List of menus to be rendered
+   *
+   * <pre className="DocPage-codeBlock">
+   * Menu: {
+   *    name: string;
+   *    label: string;
+   *    link?: string;
+   *    group?: string;
+   *    disabled?: boolean;
+   *    count?: number;
+   *    subMenu?: Menu[];
+   *    icon?: string;
+   * };
+   * </pre>
+   *
+   * | Name | Description | Default |
+   * | --- | --- | --- |
+   * | name | Menu name | |
+   * | label | Menu display name | |
+   * | link | Menu link | |
+   * | disabled | Set menu as disabled | false |
+   * | count | Value of Count (Max value 99) | |
+   * | group | Section of menu | |
+   * | subMenu | Menu submenus of type `Menu[]` | |
+   * | icon | Menu icon | |
+   */
+  menus: Menu[];
+  /**
+   * Sets menu as active
+   */
+  active?: ActiveMenu;
+  /**
+   * Callback to be called on Menu click**(only if it's not disabled)**
+   */
+  onClick?: (menu: Menu) => void;
+  /**
+   * Determines if `Navigation` expands on hover
+   */
+  hoverable?: boolean;
+  /**
+   * Makes active menu rounded **(applicable only for type: `vertical` and `expanded` menu)**
+   */
+  rounded: boolean;
+  /**
+   * Set expanded state of `Navigation`**(applicable only for type: `vertical`)**
+   */
+  expanded: boolean;
+  /**
+   * Only one SubMenu visible at a time**(applicable only for type: `vertical`)**
+   */
+  autoCollapse: boolean;
+}
+
+/**
+ * ####NOTE: VerticalMenu sets first subMenu(if present) active if the Navigation is collapsed.
+ */
+
+export const VerticalMenu = (props: VerticalMenuProps) => {
+  const {
+    menus,
+    active,
+    onClick,
+    expanded,
+    rounded,
+    autoCollapse,
+    className,
+  } = props;
+
+  const [menuState, setMenuState] = React.useState<Record<string, boolean>>({});
+  const baseProps = extractBaseProps(props);
+
+  React.useEffect(() => {
+    if (props.active) {
+      const currMenu = getMenu(menus, props.active);
+      if (currMenu) updateMenuState(currMenu, true);
+    }
+  }, [props.active]);
+
+  const updateMenuState = (menu: ActiveMenu, val?: boolean) => {
+    const currMenu = getMenu(menus, menu);
+    if (currMenu) {
+      const nameSplit = currMenu.name.split('.');
+      if (nameSplit.length > 1 || currMenu.subMenu) {
+        const name = nameSplit[0];
+        if (autoCollapse) {
+          setMenuState({ [name]: val || !menuState[name] });
+        } else {
+          const menuData = { ...menuState };
+          menuData[name] = val !== undefined ? val : !menuData[name];
+          setMenuState(menuData);
+        }
+      } else {
+        if (autoCollapse) {
+          if (!expanded) setMenuState({});
+        }
+      }
+    }
+  };
+
+  const onClickHandler = (menu: Menu) => {
+    if (!menu.disabled) {
```

Comment:
Not required if we are already adding `pointer-events: none` in css. Please verify.